src/main/java/org/kiwiproject/eureka/EmbeddedEurekaServer.java

```diff
@@ -0,0 +1,101 @@
+package org.kiwiproject.eureka;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.kiwiproject.io.KiwiPaths;
+
+/**
+ * Creates, configures, starts and stops an embedded Jetty server running a Eureka Server.
+ */
+@Slf4j
+public class EmbeddedEurekaServer {
+
+    private static final long IDLE_TIMEOUT = 3_600_000L;
+
+    private final Server eurekaServer;
+    private final ServerConnector connector;
+
+    /**
+     * Creates a new EmbeddedEurekaServer allowing Jetty to pick an available port.
+     */
+    public EmbeddedEurekaServer() {
+        this(0);
+    }
+
+    /**
+     * Creates a new EmbeddedEurekaServer with the given port.
+     *
+     * @param port the port to run Jetty on.
+     */
+    public EmbeddedEurekaServer(int port) {
+        // This is needed so Eureka starts up very quickly
+        System.setProperty("eureka.numberRegistrySyncRetries", "0");
```

Comment:
Also, if this is started by one test, will it remain set for the duration of all tests, or do you clear it somewhere (that I have not yet gotten to in this review)?

Reply:
I am not, should I do it in the stop method?

Reply:
Or maybe I do it in the extension.  Not sure anyone will be running the embedded server themselves

Reply:
That assumes stop is called right? Are we guaranteed that? We could do it there _and/or_ we could do it in the extension when all tests are complete.